llvm/lib/BinaryFormat/Dwarf.cpp

```diff
@@ -472,6 +472,117 @@ StringRef llvm::dwarf::LanguageDescription(dwarf::SourceLanguageName lname) {
   return "Unknown";
 }
 
+StringRef llvm::dwarf::LanguageDescription(dwarf::SourceLanguageName Name,
+                                           uint32_t Version) {
+  switch (Name) {
+  // YYYY
+  case DW_LNAME_Ada: {
+    if (Version <= 1983)
+      return "Ada 83";
+    if (Version <= 1995)
+      return "Ada 95";
+    if (Version <= 2005)
+      return "Ada 2005";
+    if (Version <= 2012)
+      return "Ada 2012";
```

Comment:
I'd have expected these to work the other way around - like if the Version is < 1995, then it's Ada 83. 

Might be weird to say it's Ada95 when the version is 1990/before 1995? But I'm not sure if there's a clear general sense, or language-specific sense, of what the intermediate values should mean.

Reply:
Seems like a reasonable way to deal with pre-release compilers that guess the final date of the standard incorrectly. The other alternative would be put the raw version in the string.